priv/catalog/DA114A.yaml

```diff
@@ -0,0 +1,47 @@
+id: DA114A
+name: Corosync rings
+group: Corosync
+description: |
+  Corosync has at least 2 rings configured
+remediation: |
+  ## Abstract
+  It is strongly recommended to add a second ring to the corosync communication.
+
+  ## References
+  Azure:
+
+    - https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/sap/high-availability-guide-suse-pacemaker
+
+  AWS:
+
+    - https://docs.aws.amazon.com/sap/latest/sap-hana/sap-hana-on-aws-cluster-configuration.html
+
+  GCP:
+
+    - https://cloud.google.com/solutions/sap/docs/sap-hana-ha-config-sles
+
+  SUSE / KVM:
+
+   - https://documentation.suse.com/sbp/all/single-html/SLES4SAP-hana-sr-guide-PerfOpt-15/#id-example-for-etccorosynccorosync-conf
+   - section 9.1.3 in https://documentation.suse.com/sbp/all/single-html/SLES4SAP-hana-sr-guide-PerfOpt-15/#id-adapting-the-corosync-and-sbd-configuration
+
+facts:
+  - name: corosync_nodes
+    gatherer: corosync.conf
+    argument: nodelist.node
+
+values:
+  - name: expected_corosync_nodes
+    default: 2
+  - name: expected_corosync_rings_per_node
+    default: 2
+    conditions:
+      - value: 1
+        when: env.provider == "azure" || env.provider == "gcp"
+
+expectations:
+  - name: expected_number_of_corosync_nodes
+    expect: facts.corosync_nodes.len() == values.expected_corosync_nodes
```

Comment:
I see your point, however the [current implementation](https://github.com/trento-project/runner/blob/main/runner/ansible/roles/checks/1.1.9/tasks/main.yml) does print the number of configured rings for each configured nodes ie `"22"` which is the format of the expectation  which I understand as _there should be 2 nodes and per each node 2 rings_. I might have misunderstood though.

As you said maybe we could update the description. Deal?

Reply:
I think it should update `Name`, `Description` and `Remediation`.
If it was for me, I would create 2 checks from this, because it is checking 2 different topics.
But as the check creator, up to @pirat013

Reply:
The origin check was mixing 2 checks together - expected number of nodes and expected number of corosync rings per node.
We should consider to split the check into 2 checks, which would be more 'natural' from my point of view. And we should add another check, if the related interfaces for the 2 corosync rings are configured too.